### PR TITLE
Use hpc-stack libraries in ufs-s2s-model

### LINKED ISSUE #174: Update modulefiles to use hpc-stack libraries
**Description**
Update module files on Hera and Orion and WCOSS Dell to use hpc-stack libraries.

After these updates, module files used in ufs-s2s-model and ufs-weather-model will be the same.

**Testing**
Regression tests will be done on Hera, Orion and WCOSS Dell.

### PR DESCRIPTION
**Pull request by binli2337**
### Description of changes

Update module files to use hpc-stack libraries

### Specific notes

Issues Fixed #174 

Are changes expected to change answers?
- yes

Specific changes:
- changes in conf directory (yes)
- changes in module files (yes)
- new tests added or removed (no)

Testing performed:
- machines: Hera, Orion and WCOSS Dell3

Hashes used for testing:
- NEMS: 9d05172
- CMEPS: 5e63bd4
- FV3: 6bc61df
- MOM6: 3ce81f4
- CICE: 285985c
- WW3: 96e3f3a
- FMS: f61416f
- stochastic_physics: ffdd19b

### CODE REVIEW

**Comment by binli2337** [comment]
This pull request replaces PR 184.

This pull request is ready for reviews.

File: conf/configure.fv3.hera.intel [hunk 1]
```diff
@@ -180,6 +180,6 @@ LDFLAGS += $(LIBS)
 
 ifdef InNemsMakefile
 FFLAGS += $(ESMF_INC)
-CPPFLAGS += -traditional
-EXTLIBS = $(NCEPLIBS) $(ESMF_LIB) $(LDFLAGS) $(NETCDF_LIB)
+CPPFLAGS += -DFV3_CPLD -traditional
```
Comment (junwang-noaa) [inline conf/configure.fv3.hera.intel]: May I ask why do we need to add FV3_CPLD back?

File: conf/configure.fv3.orion.intel [hunk 2]
```diff
@@ -39,7 +39,7 @@ include       $(ESMFMKFILE)
 ESMF_INC    = $(ESMF_F90COMPILEPATHS)
 
 NEMSIOINC = -I$(NEMSIO_INC)
-NCEPLIBS = $(POST_LIB) $(NEMSIO_LIB) $(G2_LIB4) $(G2TMPL_LIB) $(BACIO_LIB4) $(SP_LIBd) $(W3EMC_LIBd) $(W3NCO_LIBd) $(CRTM_LIB) $(JASPER_LIB) -lpng -lz
```
Comment (junwang-noaa) [inline conf/configure.fv3.orion.intel:42]: So is JASPER_LIB defined in hpc-stack? what is the value of it?

Comment (MinsukJi-NOAA) [reply conf/configure.fv3.orion.intel:42]: JASPER_LIB is not defined in hpc-stack.  However, JASPER_ROOT is defined.

File: conf/configure.fv3.hera.intel (hunk 1 above)
Comment (binli2337) [reply conf/configure.fv3.hera.intel]: It will be removed.

**Review by junwang-noaa — APPROVED** [review]

**Review by DeniseWorthen — APPROVED** [review]

**Comment by aerorahul** [comment]
@binli2337 
Can we use the PIO module from the hpc-stack and not build PIO as part of the S2S-model?

**Pull request merged**